test/ruby/test_method.rb

```diff
@@ -1073,8 +1091,8 @@ def test_super_method_removed
 
   def test_prepended_public_zsuper
     mod = EnvUtil.labeled_module("Mod") {private def foo; :ok end}
-    mods = [mod]
     obj = Object.new.extend(mod)
+    mods = [obj.singleton_class, mod]
```

Comment:
Interesting, so now the expected result is `[sclass, Mod, Mod0, Mod1]`, which is what would be the chain if `super` was caller in each of these methods, so that makes a lot of sense.

Reply:
It's actually `[Mod0, Mod1, sclass, Mod]`, since `Mod0` and `Mod1` are prepended to the singleton class.

Reply:
Actually, on a similar simplified example, calling these methods means only the `public` foo is called, and the private one is skipped:
```ruby
class C
  def foo
    puts caller(0), nil
    ['C']
  end
end

module Mod
  private def foo
    super + ['.']
  end
end

obj = C.new
obj.extend(Mod)

class << obj
  public :foo
end

p obj.foo
```
gives (on 3.0.3)
```
public.rb:3:in `foo'
public.rb:10:in `foo'
public.rb:21:in `<main>'

["C", "."]
```
That's I *think* because `super` doesn't consider the `owner` module as much as the "declaring module" (the one that actually `def` it), which for both sclass#foo and Mod#foo is Mod.
And there is a similar cases with aliases IIRC.

Reply:
So I think ideally `super_method` wouldn't change here, and the result of this test would be `[Mod0, Mod1, sclass]`.
That's what actually gets called if `super` is used, and it's also the result of this test on TruffleRuby.